dev/com.ibm.ws.app.manager.module/src/com/ibm/ws/app/manager/module/internal/DeployedAppInfoBase.java

```diff
@@ -512,8 +512,10 @@ protected ProtectionDomain getProtectionDomain() {
 
             ArrayList<Permission> mergedPermissions = permissionManager.getEffectivePermissions(applicationInformation.getLocation());
             int count = mergedPermissions.size();
-            for (int i = 0; i < count; i++)
-                perms.add(mergedPermissions.get(i));
+            if (count > 0) {
+                for (int i = 0; i < count; i++)
+                    perms.add(mergedPermissions.get(i));
+            }
```

Comment:
This change doesn't look like it will do anything. In the original code, if `count == 0`, the code never enters the `for` loop and nothing happens. Introducing an `if (count > 0)` check here is redundant.